### DISCUSSION TITLE
PODMAN 32-bit build

### DISCUSSION THREAD

**Discussion by sparshstark** [containers/podman#22068] [Q&A]
Hi Developers, I am recently working on a project that required to manually build podman for supporting a 32-bit device. After the build when I am trying to run an image, using the podman run command I counter this error :

OCI runtime error: crun: unknown version specified

Here are my podman details :
Client:       Podman Engine
Version:      5.0.0-dev
API Version:  5.0.0-dev
Go Version:   go1.20
Git Commit:   e38c713f3f23ebd98b7d798aa32ee92e767c4a9f
Built:        Sun Mar 17 14:58:57 2024
OS/Arch:      linux/386

/usr/bin/podman: ELF 32-bit LSB executable, Intel 80386, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux.so.2, BuildID[sha1]=a50f1a8e185440251d3e046fa48652a18659fa6e, for GNU/Linux 3.2.0, with debug_info, not stripped

This is the crun specs:
crun version 0.17
commit: 0e9229ae34caaebcb86f1fde18de3acaf18c6d9a
spec: 1.0.0
+SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL

**Comment by Luap99** [comment]
you need a newer crun (https://github.com/containers/crun/releases/tag/1.14.3)

**Reply by arogozhnikov** [reply]
I have 5.0.1 from brew, and crun 1.4.4 (also from brew), but still getting this error. Are you sure about this being the reason?
```
crun --version
crun version 1.14.4
commit: a220ca661ce078f2c37b38c92e66cf66c012d9c1
rundir: /run/user/1001/crun
spec: 1.0.0
+SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL

podman --version
podman version 5.0.1
```

**Reply by Luap99** [reply]
yes 100%, you need to check `podman info` to see what podman is using.

**Reply by rhatdan** [reply]
The crun is inside of the machine, crun on the Mac does not matter.

**Reply by arogozhnikov** [reply]
I use brew on linux.

**Reply by arogozhnikov** [reply]
I've made a number of strange and likely unrelated steps in my env, and brew's podman 5 works now (with --net=host as otherwise complains on pasta)

**Reply by sbrivio-rh** [reply]
This is probably coming from the fact that two different versions of `crun` are installed, one via Homebrew, and the other one via Ubuntu package, see also https://github.com/Homebrew/homebrew-core/issues/173699#issuecomment-2283403738.

**Comment by rhatdan** [comment]
@giuseppe PTAL

**Reply by YevheniiPokhvalii** [reply]
I have the same problem - I described it at the end of this github issue: https://github.com/Homebrew/homebrew-core/issues/173699

**Comment by DDilinger** [comment]
dnf install crun-1.14.3
